test/FunctionalTests/Microsoft.OData.Core.Tests/MessageWriterConcurrencyTests.cs

```diff
@@ -0,0 +1,78 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MessageWriterConcurrencyTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.IO;
+using System.Threading.Tasks;
+using System;
+using Xunit;
+using Microsoft.Extensions.DependencyInjection;
+using System.Linq;
+using Microsoft.OData.Tests;
+
+namespace Microsoft.OData.Core.Tests
+{
+    public class MessageWriterConcurrencyTests
+    {
+        /// <summary>
+        /// Verifies that concurrent message writer does not interleave execution and isolates the underlying streams.
+        /// </summary>
+        /// <returns>A task for the asynchronous test</returns>
+        [Fact]
+        public async Task VerifyConcurrentResultsAreIsolatedAsync()
+        {
+            ServiceCollection services = new();
+            services.AddDefaultODataServices();
+            ServiceProvider serviceProvider = services.BuildServiceProvider();
+
+            await Task.CompletedTask; // Added due to dotnet < 5 as async await cannot be used only in a loop

```

Comment:
Do we need this? The target framework for Microsoft.OData.Core is net8.0

Reply:
@marabooy Which projects are these? Are they loaded when you open _OData.sln_?

Reply:
I believe these are the tests let me find them and share here.

Reply:
https://github.com/OData/odata.net/blob/main/buildandtest.yml#L24 This line when building will compile vs an older version and causes the failure

Reply:
@marabooy Could you try moving this task https://github.com/OData/odata.net/blob/95233fc8a399cbae6e816216528d02a581b77b48/buildandtest.yml#L24 to here https://github.com/OData/odata.net/blob/95233fc8a399cbae6e816216528d02a581b77b48/nightly.yml#L14 (since that's the task that require .NET 2.1). Let's see if the build pipeline runs okay and whether you can drop this line.